docs/source/architecture.rst

```diff
@@ -0,0 +1,118 @@
+Architectural overview
+======================
+
+Background
+----------
+`HashiCorp Vault <https://www.vaultproject.io/>`_ provides a secure
+clearinghouse for sensitive information required by our applications, including
+engines for creating, managing, and revoking short-lived credentials for AWS
+resources, databases, and other protected systems. Using Vault increases our
+overall security profile by eliminating the need to embed secrets in
+application configurations, and makes it much easier to implement short
+lifetimes and rotation without a ton of manual human intervention.
+
+Applications need to be able to obtain sensitive information from Vault
+via its API. Handling this outside of the application is a bit janky; either a
+sidecar process is required, or the app must be periodically killed and
+restarted with fresh secrets (e.g. database credentials).
+
+The goal of this project is to reduce the complexity of using Vault secrets by
+implementing a lightweight Python integration that runs inside the client app.
```

Comment:
This should probably include "lightweight Python integration using HashiCorp's python library for Valut, HVAC..."